File: src/ProductionMES.Domain/Entities/Permission.cs
using ProductionMES.Domain.Enums;

namespace ProductionMES.Domain.Entities;

/// <summary>
/// Catalog cố định các permission hợp lệ trong hệ thống (ADR-004) — mỗi bản ghi là 1 cặp
/// (<see cref="Resource"/>, <see cref="Action"/>) khớp đúng action thật đang tồn tại ở Controller tương ứng.
/// KHÔNG phải tích chéo đầy đủ Resource x Action (vd. <c>Line</c> không có <c>Delete</c> vì API chỉ có
/// <c>deactivate</c>, không xóa cứng) — xem seed đầy đủ ở <c>DbSeeder</c>. Bảng này không có API xóa/sửa,
/// chỉ <c>RolePermission</c> (cấp/thu hồi permission cho role) mới được Admin chỉnh qua UI runtime.
/// </summary>
public class Permission
{
    public int Id { get; set; }

    public PermissionResource Resource { get; set; }

    public PermissionAction Action { get; set; }
}

File: web-admin/src/types/permission.ts
/**
 * Type khớp DTO Permission backend (`ProductionMES.Application/DTOs/Permissions/`) và enum
 * `ProductionMES.Domain/Enums/{PermissionResource,PermissionAction}.cs` (ADR-004).
 */
import type { UserRole } from './auth';

/** Khớp enum `PermissionResource` backend, serialize dạng chuỗi (mục 10 API-Conventions.md). */
export type PermissionResource = 'Line' | 'Stage' | 'WorkStation' | 'ProductionPlan' | 'ProductionPlanStage';

/** Khớp enum `PermissionAction` backend, serialize dạng chuỗi (mục 10 API-Conventions.md). */
export type PermissionAction = 'View' | 'Create' | 'Update' | 'Activate' | 'Deactivate' | 'Delete';

/** Khớp `PermissionDto` (`ProductionMES.Application/DTOs/Permissions/PermissionDto.cs`). */
export interface Permission {
  id: number;
  resource: PermissionResource;
  action: PermissionAction;
}

/**
 * Khớp `RolePermissionMatrixDto` (`ProductionMES.Application/DTOs/Permissions/RolePermissionMatrixDto.cs`)
 * — 1 dòng cho mỗi `UserRole`, kèm danh sách `id` permission (khớp `Permission.id`) đang được cấp.
 */
export interface RolePermissionMatrixEntry {
  role: UserRole;
  permissionIds: number[];
}
